File: 66_Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        return plusOneHelper(digits, digits.size() - 1); 
    }
    
    vector<int> plusOneHelper(vector<int> digits, int index){
        if(index < 0){
            auto it = digits.insert(digits.begin(), 1); 
            return digits; 
        }
        if(digits[index] + 1 == 10){
            digits[index] = 0;
            return plusOneHelper(digits, index - 1); 
        }
        else{
            digits[index]++; 
        }
        return digits; 
    }
};

File: 66_Plus_One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        return self.plusOneHelper(digits, len(digits) - 1) 
    
    def plusOneHelper(self, digits, index):
        if index < 0:
            digits.insert(0,1) 
            return digits 
                
        if(digits[index] + 1 == 10): 
            digits[index] = 0 
            return self.plusOneHelper(digits, index - 1) 
        else:
            digits[index] += 1 
        return digits 